msprime/simulations.py

```diff
@@ -1343,3 +1344,33 @@ def print_history(self, output=sys.stdout):
             print("=" * len(s), file=output)
             self._print_populations(epoch, output)
             print(file=output)
+
+    def popn_size_history(self):
+        """
+        Returns a matrix of population sizes seen
+        by samples from each population at each
+        time defined by the epochs.
+        """
+        num_pops = len(self.epochs[0].populations)
+        pop_size = numpy.zeros((num_pops, len(self.epochs)))
+        popn_set = [[x] for x in range(num_pops)]
+        for j, epoch in enumerate(self.epochs):
+            for k, pop in enumerate(epoch.populations):
+                idx = [k in x for x in popn_set].index(True)
+                pop_size[k, j] = epoch.populations[idx].start_size
+            for event in epoch.demographic_events:
```

Comment:
Hmmm, I'm not convinced about this. The population size isn't actually zero, it's just that there aren't any samples in there. This seems like an unnecessary complication to me. (If you want to set the population size of zero you can do this in the model explicitly with an extra event)

Reply:
not sure i follow here- are you referring to the fact that i'm initializing the population sizes to zero in the pop_size array? Or to when i'm iterating in the loop on 1358 over populations present in the given epoch?

Reply:
No, I'm objecting to looping over the demographic events and setting the population size to zero if there was a full MassMigration. This seems like an unnecessary complication to me, and probably isn't safe in any case --- what if there was still migration into this deme?

Reply:
ah I see. No population sizes in the history object are actually getting set to zero except at initialization as I'm setting by `epoch.populations[idx].start_size` which should always be positive. I am however ignoring the case where there has been a Mass Migration with proportion = 1 (i.e. a back in time pop merger) followed by subsequent migration back in to the source population. This case would make sense biologically, but I can see that the code would not be safe.

Thoughts about a better way to do this? The goal is slightly strange-- I want to know the population size history experienced by samples tracing back in time through population merger events.

Reply:
If we adopt the convention that any 'closed' population should have its population size set to zero by the user, then there's no need to have any special cases in here. I'm probably misinterpreting the goal here though. My understanding here is that the population size of the deme in question is effectively zero because there was a population merger, and we're trying to show this. Is this right or am I misunderstanding?